google-ads/src/test/java/com/google/ads/googleads/lib/callables/ExceptionTransformingServerStreamingCallableTest.java

```diff
@@ -0,0 +1,103 @@
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.ServerStreamingCallable;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.grpc.StatusException;
+import java.util.Objects;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ExceptionTransformingServerStreamingCallableTest {
+
+  @Mock public ResponseObserver<Object> innerObserver;
+  private MockServerStreamingCallable innerCallable;
+
+  private static final GoogleAdsFailure googleAdsFailure =
+      GoogleAdsFailure.newBuilder().addErrors(GoogleAdsError.getDefaultInstance()).build();
+  private final Object request = "test";
+  private final ApiCallContext context = GrpcCallContext.createDefault();
+  private ExceptionTransformingServerStreamingCallable<Object, Object> serverStreamingCallable;
+  private ApiException rawApiException;
+
+  @Before
+  public void setup() {
+    this.innerCallable = new MockServerStreamingCallable();
+    this.serverStreamingCallable =
+        new ExceptionTransformingServerStreamingCallable(
+            innerCallable, new GoogleAdsExceptionTransformation());
+
+    Metadata trailers = new Metadata();
+    trailers.put(
+        GeneratedCatalog.getDefault().getLatestVersion().getExceptionFactory().getTrailerKey(),
+        googleAdsFailure.toByteArray());
+
+    rawApiException =
+        new ApiException(
+            new StatusException(Status.INVALID_ARGUMENT, trailers),
+            GrpcStatusCode.of(Code.INVALID_ARGUMENT),
+            false);
+  }
+
+  @Test
+  public void transformsExceptions() {
+    serverStreamingCallable.call(request, innerObserver, context);
+    innerCallable.setException(rawApiException);
+    verify(innerObserver).onError(argThat(new GoogleAdsExceptionMatcher(googleAdsFailure)));
+  }
+
+  private static class GoogleAdsExceptionMatcher extends ArgumentMatcher<GoogleAdsException> {
+
+    private final GoogleAdsFailure googleAdsFailure;
+
+    public GoogleAdsExceptionMatcher(GoogleAdsFailure googleAdsFailure) {
+      this.googleAdsFailure = googleAdsFailure;
+    }
+
+    @Override
+    public boolean matches(Object argument) {
+      if (argument == null) {
+        return false;
+      }
+      if (!argument.getClass().isAssignableFrom(GoogleAdsException.class)) {
+        return false;
+      }
+      GoogleAdsException exception = (GoogleAdsException) argument;
+      return Objects.equals(googleAdsFailure, exception.getGoogleAdsFailure());
+    }
+  }
+
+  private static class MockServerStreamingCallable<RequestT, ResponseT>
+      extends ServerStreamingCallable<RequestT, ResponseT> {
+
+    private volatile ResponseObserver<ResponseT> responseObserver;
```

Comment:
nit: Is there any multithreading happening that makes this `volatile` necessary?

Reply:
Nope, removed.